opentelemetry/proto/metrics/v1/metrics.proto

```diff
@@ -87,93 +88,121 @@ message InstrumentationLibraryMetrics {
+    // ValueTypes use the odd numbers.
+
+    // ScalarInt64 implies that Metric.Points MUST use DataPoint.value_int64.
+    SCALAR_INT64     = 2;
+    // ScalarDouble implies that Metric.Points MUST use DataPoint.value_double.
+    SCALAR_DOUBLE    = 3;
+
+    // HistogramInt64 implies that Metric.Points MUST use
+    // DataPoint.histogram with int64 boundaries.
+    HISTOGRAM_INT64  = 4;
+    // HistogramDouble implies that Metric.Points MUST use
+    // DataPoint.histogram with floating point boundaries.
+    HISTOGRAM_DOUBLE = 5;
+
+    // SummaryInt64 implies that Metric.Points MUST use DataPoint.summary
+    // with int64 min/max/sum/last/count values.
+    SUMMARY_INT64    = 6;
+    // SummaryDouble implies that Metric.Points MUST use
+    // DataPoint.summary with double min/max/sum/last/count values.
+    SUMMARY_DOUBLE   = 7;
+
+    // RawInt64 implies that Metric.Points MUST use DataPoint.raw with
+    // int64 values.
+    RAW_INT64        = 8;
+    // RawDouble implies that Metric.Points MUST use DataPoint.raw with
+    // double value.
+    RAW_DOUBLE       = 9;
   }
 
-  // type is the type of values this metric has.
-  Type type = 4;
+  // ValueType is the type of values this metric has.
+  ValueType value_type = 4;
 
-  // Temporality is the temporal quality values of a metric have. It
-  // describes how those values relate to the time interval over which they
-  // are reported.
-  enum Temporality {
-    // INVALID_TEMPORALITY is the default Temporality, it MUST not be
-    // used.
-    INVALID_TEMPORALITY = 0;
+  // KindElement contains a set of bit masks used to construct Kind enum values.
+  // There are 7 bits used presently, broken into groups:
+  //
+  //   Temporality is the temporal quality values of a metric have. It
+  //   describes how those values relate to the time interval over
+  //   which they are reported.  One of the 3 Temporality values
+  //   (CUMULATIVE, DELTA, INSTANTANEOUS) MUST be set.
+  //
+  //   Structure indicates whether metric data describes a sum of
+  //   measurements (ADDING) or a collection of individual
+  //   measurements (GROUPING).  Value types may be interpreted
+  //   differently depending on Structure.  For example, a Histogram
+  //   DataPoint may be computed for a Counter (ADDING) instrument
+  //   or a ValueRecorder (GROUPING) instrument, and:
+  //
+  //   - ADDING Histograms define a meaningful Sum in the form of a
+  //     distribution of constituent measurements.
+  //   - GROUPING Histograms define a distribution of measurements,
+  //     where the sum is potentially not meaningful.  If also
+  //     SYNCHRONOUS, these data points define a rate in terms of
+  //     their Count.
+  //
+  //   Monotonicity applies only to ADDING data points.  When set,
+  //   MONOTONIC indicates that the calculated sum can be monitored
+  //   as a rate of change.
+  //
+  //   Synchronicity applies to data points produced when the
```

Comment:
How is this consumed by a backend? If the data are coming from other library like Prometheus how would I know what to set here? They define exemplars as well and I don't know in advance if a metric will or not have exemplars associated (in the collector). Is this something that we really need or we can remove to not complicate translation from other Protocols.

Reply:
I rewrote this as:
```
  //   Synchronicity applies to data points produced when the
  //   application called the API, as opposed to the SDK calling the
  //   application via a callback.  When set, SYNCHRONOUS indicates
  //   that there may be an associated trace context.  The rate of
  //   synchronous events has a meaningful interpretation, when
  //   considering the DataPoint.Histogram.Count or
  //   DataPoint.Summary.Count fields, as these are true
  //   application-level events.
  //
  //   Data points without SYNCHRONOUS are considered asynchronous.
  //   The DataPoint.Histogram.Count and DataPoint.Summary.Count
  //   fields record a number of observations, which is controlled
  //   both by the number of distinct label sets observed and by the
  //   rate of collection.  Because asynchronous metric Count fields
  //   depend on the rate of collection, the rate of these metrics
  //   should not be interpreted as rates (however these Counts may
  //   still be used to form an average value of these metrics).
```

Reply:
> If the data are coming from other library like Prometheus how would I know what to set here? 

Hopefully SYNCHRONOUS is always a safe option. This makes me think that the optional flag should be ASYNCHRONOUS, and it will only be set when for sure the value was generated by a variable-rate callback controlled by the SDK, not user-level events. How's this sound?

Reply:
It feels to me that this information cannot be guaranteed, and would like to challenge if this is an important and well defined property that should be in the protocol. Maybe consider to be removed instead

Reply:
If we cannot include this information, the entire part of the specification that deals with asynchronous instrument semantics was for nothing. I think when ASYNCHRONOUS is true it is guaranteed.